### ISSUE TITLE
Issue: Update-OMEFirmware with device filter option

### ISSUE THREAD

**Issue by dgobji** [dell/OpenManage-PowerShell-Modules#15]
We have script in place which upgrades firmware of specific server using Update-OMEFirmware and uses devicefilter to apply it to specific server. But This device filter works only if the server is present on OME and if doesnt it basically upgrades each and every server.
We faced issue where our whole vmware setup went into firmware upgrade state resulting a downtime of 1 hour

**Comment by TrevorSquillario** [comment]
I'll have a look. Can you send me the code snipped you were using if possible, Thanks!

**Comment by dgobji** [comment]
Update-OMEFirmware -Baseline $firmwarebaseline -DeviceFilter $vmhead_ome -UpdateSchedule "RebootNow" -ClearJobQueue @defaultProps

Here in our badluck $vmhead_ome was null and we end up upgrading firmware for all the ESXi at the same time.

It will be best if we can have firmware upgrade command which works directly on Server device

**Comment by TrevorSquillario** [comment]
If `-DeviceFilter` is $null then it updates all the hosts in the baseline. It's $null if you don't specify `-DeviceFilter` or if there are no devices returned. You could do a $null check on `$vmhead_ome` in your code but that leaves folks open to the issue you experienced which isn't good at all. I need to think on this for a bit...

**Comment by dgobji** [comment]
@TrevorSquillario Thnks for the update, Yes for now we have added condition to make sure $vmhead_ome is not null,
But, Having said that this seem to be an issue as it has big impact if the value is null and one can end up upgrading all the Node without maintenance window

**Comment by TrevorSquillario** [comment]
I think I found a fix. You can use `$PSBoundParameters` to check if the parameter is specified in the script. So if it's specified AND it's null I can throw and exception and kill the script execution. Let me do some more testing and I'll get back to you on this.

**Comment by dgobji** [comment]
Thanks Trevor, Will for your update

**Comment by TrevorSquillario** [comment]
It will now throw an error `-DeviceFilter is specified but contains no devices` if you run `Update-OMEFirmware` with a `-DeviceFilter` and it's null. You can set `-UpdateSchedule "Preview"` to view the list of updates before applying them as a sanity check for the future. Sorry for any inconvenience this may have caused. 

```
Update-OMEFirmware -Baseline $("Test512" | Get-OMEFirmwareBaseline) -DeviceFilter $("C86C0Q2", "C86C0QZ" | Get-OMEDevice -FilterBy "ServiceTag") -UpdateSchedule "Preview" -RebootType "Graceful" -Verbose
```

Fixed in Release 3.6.2
https://github.com/dell/OpenManage-PowerShell-Modules/releases/tag/3.6.2
